### PR TITLE
Sm/project subsumes schema

### PR DESCRIPTION
**Pull request by saurabmitra3**
exposes only project command
get all projects and details
export project schemas and content from Mason into a folder on your local
import project and/or schemas from project folder into Mason

### CODE REVIEW

File: mason/mason.go [hunk 1]
```diff
@@ -204,6 +311,9 @@ func getSchema(schemaId string, all bool, output string) {
 	if schemaId != "" {
 		q.Add("schema_id", schemaId)
 	}
+	if projectId != "" {
```
Comment (mercurialsolo) [inline mason/mason.go:287]: Please do a check for type as well as null

File: mason/mason.go [hunk 2]
```diff
@@ -59,10 +59,83 @@ func writeFileJson(path string, content []byte) bool {
 	return true
 }
 
-func getProjects(projectId string, all bool, output string) {
+func exportProject(projectId string, output string) {
+	var outputSchema = ""
+	var outputContent = ""
+	var projectPath = ""
+	if output != "" {
```
Comment (mercurialsolo) [inline mason/mason.go:66]: check for null

File: mason/mason.go [hunk 3]
```diff
@@ -59,10 +59,83 @@ func writeFileJson(path string, content []byte) bool {
...
+		fmt.Printf("Wrote Project Schema and Content details to %s\n", projectPath)
+	}
+}
+
+func importProject(projectPath string, schemaOnly bool, contentOnly bool) {
+	if !schemaOnly && !contentOnly {
+		schemaOnly = true
+		contentOnly = true
+	}
```
Comment (mercurialsolo) [inline mason/mason.go:93]: What is this doing? Are we saying we will accept any values in schemaOnly / contentOnly

**Review by mercurialsolo — COMMENTED** [review]
Some defensive checks need to be done and redundant code needs elimination.

File: mason/mason.go (hunk 1 above)
Comment (saurabmitra3) [reply mason/mason.go:287]: By default from Flagset we will get this as `""` if it is null

File: mason/mason.go (hunk 2 above)
Comment (saurabmitra3) [reply mason/mason.go:66]: same as above

File: mason/mason.go (hunk 3 above)
Comment (saurabmitra3) [reply mason/mason.go:93]: There was some issue parsing schemaOnly and contentOnly as default `true` in flagset.
That's why this, if no option is sent, both `schema` and `content` are imported

**Review by karthik-varma04 — APPROVED** [review]

**Pull request merged**